### PR TITLE
SetController test

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/procedures/__tests__/SetController.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    });
+  });
+
+  describe('SetController', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate SetController with incorrect security symbol
+      target = new SetController(
+        {
+          symbol: params1.symbol,
+          controller: params1.controller,
```
Comment (victorwiebe) [inline src/procedures/__tests__/SetController.ts]: Instantiate please with just params (I’ve gotten recent review from Jere about this type of issue)

Comment (victorwiebe) [reply src/procedures/__tests__/SetController.ts]: Simplify the code just passing params object in

File: src/procedures/__tests__/SetController.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+      // Real call rejects
+      expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `You must be the owner of this Security Token to set the controller`,
+        })
+      );
+    });
+
```
Comment (victorwiebe) [inline src/procedures/__tests__/SetController.ts]: Description should be rewritten like:
”should add a transaction to the queue to set a controller on the security token”

File: src/procedures/__tests__/SetController.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  let contextMock: MockManager<contextModule.Context>;
+  let wrappersMock: MockManager<wrappersModule.PolymathBase>;
+  let tokenFactoryMock: MockManager<
+    tokenFactoryModule.MockedTokenFactoryObject
+  >;
+  let securityTokenMock: MockManager<
+    contractWrappersModule.SecurityToken_3_0_0
+  >;
+
+  let tokenFactoryMockStub: SinonStub<any, any>;
```
Comment (victorwiebe) [inline src/procedures/__tests__/SetController.ts]: This line is likely not needed

File: src/procedures/__tests__/SetController.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params.symbol}`,
+        })
+      );
+    });
+
+    test('should call error on inappropriate params address', async () => {
+      // Instantiate SetController with incorrect args instead
+      target = new SetController(
+        {
```
Comment (monitz87) [inline src/procedures/__tests__/SetController.ts]:
```
{
  ...params,
  controller: 'Inappropriate',
}
```

**Pull request merged**